src/TurbulenceAveragingPostProcessing.C

```diff
@@ -735,6 +869,233 @@ TurbulenceAveragingPostProcessing::compute_favre_stress(
+  stk::mesh::FieldBase *TurbKe_ = metaData.get_field(stk::topology::NODE_RANK, "turbulent_ke");
+  if(TurbKe_ == NULL) computeSFSTKE = true ;
+  stk::mesh::FieldBase *DualNodalVolume_ = metaData.get_field(stk::topology::NODE_RANK, "dual_nodal_volume");
+  stk::mesh::FieldBase *DuDx_ = metaData.get_field(stk::topology::NODE_RANK, "dudx");
+  stk::mesh::FieldBase *SFSStress = metaData.get_field(stk::topology::NODE_RANK, SFSStressFieldName);
+
+  stk::mesh::BucketVector const& node_buckets_sfsstress =
+    realm_.get_buckets( stk::topology::NODE_RANK, s_all_nodes );
+  for ( stk::mesh::BucketVector::const_iterator ib = node_buckets_sfsstress.begin();
+        ib != node_buckets_sfsstress.end() ; ++ib ) {
+    stk::mesh::Bucket & b = **ib ;
+    const stk::mesh::Bucket::size_type length   = b.size();
+
+    // fields
+    const double * turbNu_ = (double*)stk::mesh::field_data(*TurbViscosity_, b);
+    double * turbKe_;
+    if (!computeSFSTKE)
+        turbKe_ = (double*)stk::mesh::field_data(*TurbKe_, b);
+    const double * dualNodalVolume_ = (double*)stk::mesh::field_data(*DualNodalVolume_, b);
+    const double * dudx_ = (double*)stk::mesh::field_data(*DuDx_, b);
+    double *sfsstress_ = (double*)stk::mesh::field_data(*SFSStress,b);
+    const int offSet = 6;
+
+    // Store the xx, xy, xz, yy, yz, zz components of sfs_stress
+    for ( stk::mesh::Bucket::size_type k = 0 ; k < length ; ++k ) {
+      double divU = 0.0;
+      for ( int j = 0; j < nDim; ++j)
+          divU += dudx_[k*offSet+(nDim*j+j)] ;
+      double sfstke = 0.0;
+      if(computeSFSTKE) {
+          // Use method of Yoshisawa (1986) - Statistical theory for compressible turbulent shear flows, with the application to subgrid modeling, 29, 2152.
+          double Ci = 0.9;
+          double sijMag = 0.0;
+          for ( int i = 0; i < nDim; ++i ) {
+              for ( int j = 0; j < nDim; ++j ) {
+                  const double rateOfStrain = 0.5*(dudx_[k*offSet+nDim*i+j] + dudx_[k*offSet+nDim*j+i]);
+                  sijMag += rateOfStrain*rateOfStrain;
+              }
+          }
+          sijMag = std::sqrt(2.0*sijMag);
+          sfstke = 2.0 * Ci * std::pow(dualNodalVolume_[k], 2.0*invNdim) * sijMag;
+      } else {
+          sfstke = turbKe_[k];
+      }
+      size_t componentCount = 0;
+      for ( int i = 0; i < nDim; ++i ) {
+          for ( int j = i; j < nDim; ++j ) {
+              const double divUTerm = ( i == j ) ? 2.0/3.0*divU : 0.0;
+              const double sfsTKEterm = ( i == j ) ? 2.0/3.0*sfstke : 0.0;
+              const double newStress = (sfsstress_[k*offSet + componentCount]*oldTimeFilter*zeroCurrent - dt*(turbNu_[k]*(dudx_[k*offSet+(nDim*i+j)] + dudx_[k*offSet+(nDim*j+i)] - divUTerm) - sfsTKEterm))/currentTimeFilter_ ;
+              sfsstress_[k*offSet + componentCount] = newStress;
+              componentCount++;
+          }
+      }
+    }
+  }
+}
+
+
+void
+TurbulenceAveragingPostProcessing::compute_temperature_sfs_stress(
```

Comment:
Maybe call this the``sfs temperature flux"?  It's not a stress, really.  Same with the resolved portion.

Reply:
@gantech If this is being renamed, also rename the input file options.

Reply:
Is SFS a term used in atmospheric sciences? In general, I would like to see a generalized approach so that we could compute, for example, any \bar{rho*u'\phi'}. Would that be too hard? Otherwise, we will need to generalize a bit later. I could use this capability in a variety of projects...